### ISSUE TITLE
0.3.0 bug: Queries on 'posts' with category value return internal server error

### ISSUE THREAD

**Issue by paulisloud** [wp-graphql/wp-graphql#788]
This is fine:
```
query GET_POSTS {
  posts(first: 1) {
    edges {
      node {
        id
        title
        tags {
          nodes {
            name
          }
        }
      }
    }
  }
}
```

This returns a query result with an error:

```
query GET_POSTS {
  posts(first: 1) {
    edges {
      node {
        id
        title
        categories {
          nodes {
            name
          }
        }
      }
    }
  }
}
```

A sample result:

```
{
  "errors": [
    {
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 8,
          "column": 11
        }
      ],
      "path": [
        "posts",
        "edges",
        0,
        "node",
        "categories",
        "nodes",
        0
      ]
    }
  ],
  "data": {
    "posts": {
      "edges": [
        {
          "node": {
            "id": "xyz123",
            "title": "The post title",
            "categories": {
              "nodes": [
                null
              ]
            }
          }
        }
      ]
    }
  }
}
```

**Comment by epeli** [comment]
Can you add `define( 'GRAPHQL_DEBUG', true );` to your `wp-config.php` to get a better error message?

**Comment by paulisloud** [comment]
Sure:

```
{
      "debugMessage": "Method WPGraphQL\\Data\\Loader\\TermObjectLoader::loadKeys is expected to return an array with keys but got: null",
      "message": "Internal server error",
...
```

**Comment by jasonbahl** [comment]
Sweet. Thanks! The debug message helps. I know what to fix now.

**Comment by jasonbahl** [comment]
@paulisloud I'm not replicating this. Can you share some example data so I can try and get to the same point?

Perhaps we can figure out what the case is causing this and make sure we get a test in place to prevent it from happening again. 

When I look at the code that resolves this conneciton, I'm not seeing anything obvious that would be causing this issue. 

Going on a limb, but is there any chance the Post you are querying is associated with some Categories that have been deleted? 🤔 

----

Here are some attempts at duplicating this issue:

**Query a Post with No Categories assigned to it**

![Screen Shot 2019-04-08 at 12 41 07 PM](https://user-images.githubusercontent.com/1260765/55748391-c4819200-59fb-11e9-8094-6c8f070116ca.png)

**Query a Post with one Category assigned to it**

![Screen Shot 2019-04-08 at 12 41 26 PM](https://user-images.githubusercontent.com/1260765/55748392-c4819200-59fb-11e9-8a30-cc1bdd403d77.png)

**Query a Post with more than one Category assigned to it**

![Screen Shot 2019-04-08 at 12 41 49 PM](https://user-images.githubusercontent.com/1260765/55748394-c4819200-59fb-11e9-930b-d5601d30e161.png)

**Query a Post with No Tag assigned to it**

![Screen Shot 2019-04-08 at 12 39 53 PM](https://user-images.githubusercontent.com/1260765/55748388-c3e8fb80-59fb-11e9-982c-bac2b1c4d0b4.png)

**Query a Post with one Tag assigned to it**

![Screen Shot 2019-04-08 at 12 40 10 PM](https://user-images.githubusercontent.com/1260765/55748389-c3e8fb80-59fb-11e9-8f3d-b7b2d4a0ab56.png)

**Query a Post with more than one Tag assigned to it**

![Screen Shot 2019-04-08 at 12 40 25 PM](https://user-images.githubusercontent.com/1260765/55748390-c4819200-59fb-11e9-80f1-47857fa88193.png)

**Comment by paulisloud** [comment]
Hi @jasonbahl, just pulled 0.3.01 and the issue is fixed ¯\_(ツ)_/¯

Thanks for the quick attention to this, whatever you did!

**Comment by paulisloud** [comment]
Looking over TermObjectLoader::loadKeys, the `get_term_by` looks up based on `term_taxonomy_id`, which seems to me based on the values being passed in that it should be `term_id`. This is causing mismatches in results. I.e. posts responding to queries with categories they don't have. The line in question:
```
$term_object                = get_term_by( 'term_taxonomy_id', $key ); 
```

**Comment by jasonbahl** [comment]
@paulisloud can you confirm this is fixed now with PR #824?

I'll write a test for it tomorrow to prevent regressions if you can confirm.

**Comment by paulisloud** [comment]
Roger, we're golden.
